core/src/main/java/org/apache/calcite/sql/SqlMerge.java

```diff
@@ -205,32 +229,37 @@ public void setSourceSelect(SqlSelect sourceSelect) {
     writer.keyword("ON");
     condition.unparse(writer, opLeft, opRight);
 
-    SqlUpdate updateCall = this.updateCall;
-    if (updateCall != null) {
-      unparseUpdateCall(writer, opLeft, opRight);
-    }
-
-    SqlDelete deleteCall = this.deleteCall;
-    if (deleteCall != null) {
-      unparseDeleteCall(writer, opLeft, opRight);
-    }
-
-    SqlInsert insertCall = this.insertCall;
-    if (insertCall != null) {
-      writer.newlineAndIndent();
-      writer.keyword("WHEN NOT MATCHED");
-      if (this.insertCall instanceof SqlMergeInsert
-          && ((SqlMergeInsert) this.insertCall).condition != null) {
-        writer.keyword("AND");
-        ((SqlMergeInsert) this.insertCall).condition.unparse(writer, opLeft, opRight);
-      }
-      writer.keyword("THEN INSERT");
-      SqlNodeList targetColumnList = insertCall.getTargetColumnList();
-      if (targetColumnList != null) {
-        targetColumnList.unparse(writer, opLeft, opRight);
+    List<SqlCall> callOrderList = this.getCallOrderList();
+    for (SqlCall call: callOrderList) {
+      if (call instanceof SqlUpdate) {
+        SqlUpdate updateCall = this.updateCall;
+        if (updateCall != null) {
```

Comment:
no need to have a extra variable `updateCall`. You can directly check in if condition
`this.updateCall != null`

Reply:
Directly used this.updateCall != null in if condition